bluesky/callbacks.py

```diff
@@ -392,6 +351,147 @@ def stop(self, stop_document):
+    rowwise : bool
+        If True, print each row to stdout. If False, reprint the full updated
+        table each time. This is useful if other messsages are interspersed.
+    print_header_interval : int
+        The number of events to process and print their rows before printing
+        the header again
+    use_filestore : bool
+        Query filestore for external data
+
+    Examples
+    --------
+    Show a table with motor and detector readings as the events are generated.
+
+    >>> RE(stepscan(motor, det), LiveTable(['motor', 'det']))
+    +------------+-------------------+----------------+----------------+
+    |   seq_num  |             time  |         motor  |   sum(det_2d)  |
+    +------------+-------------------+----------------+----------------+
+    |         1  |  12:46:47.503068  |         -5.00  |        449.77  |
+    |         3  |  12:46:47.682788  |         -3.00  |        460.60  |
+    |         4  |  12:46:47.792307  |         -2.00  |        584.77  |
+    |         5  |  12:46:47.915401  |         -1.00  |       1056.37  |
+    |         7  |  12:46:48.120626  |          1.00  |       1056.50  |
+    |         8  |  12:46:48.193028  |          2.00  |        583.33  |
+    |         9  |  12:46:48.318454  |          3.00  |        460.99  |
+    |        10  |  12:46:48.419579  |          4.00  |        451.53  |
+    +------------+-------------------+----------------+----------------+
+
+
+    """
+    def __init__(self, *args, print_header_interval=50, rowwise=True,
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.rowwise = rowwise
+        self.num_events_since_last_header = 0
+        self.print_header_interval = print_header_interval
+
+    def create_table(self):
+        super().create_table()
+
+        if self.rowwise:
+            self._print_table_header()
+        sys.stdout.flush()
+
+    def _print_table_header(self):
+        print('\n'.join(str(self.table).split('\n')[:3]))
+
+    def start(self, start_document):
+        super().start(start_document)
+        self.create_table()
+
+    def event(self, event_document):
+        row = self.document_to_row(event_document)
+        with self._lock:
+            self.table.add_row(row)
+
+            if self.rowwise:
+                # Print the last row of data only.
+                # [-1] is the bottom border
+                print(str(self.table).split('\n')[-2])
+                # only print header intermittently for rowwise table printing
+                if self.num_events_since_last_header >= self.print_header_interval:
+                    self._print_table_header()
+                    self.num_events_since_last_header = 0
+                self.num_events_since_last_header += 1
+            else:
+                # print the whole table
+                print(self.table)
+
+            sys.stdout.flush()
+
+
+class PostRunTable(TableBase):
```

Comment:
It definitely make sense to do it that way. A couple reasons why I chose this way as a first pass are:
* Re-using most of the `LiveTable` implementation made this easy
* "Collect data, close run, write to db, read from db, print to screen" is a rather inefficient way of doing things when you have most of the data (as you say, it can be lossy) already in memory. Perhaps once the caching layer is implemented on the db side doing this won't be such an issue.